bot/exts/utilities/githubinfo.py

```diff
@@ -293,78 +307,145 @@ async def github_user_info(self, ctx: commands.Context, username: str) -> None:
 
         await ctx.send(embed=embed)
 
+    @tasks.loop(hours=24)
+    async def refresh_repos(self) -> None:
+        """Refresh self.repos with latest PyDis repos."""
+        self.repos, _ = await self.fetch_data(REPOSITORY_ENDPOINT.format(org="python-discord"))
+        log.info(f"Loaded {len(self.repos)} repos from Python Discord org into memory.")
+
     @github_group.command(name="repository", aliases=("repo",))
     async def github_repo_info(self, ctx: commands.Context, *repo: str) -> None:
         """
-        Fetches a repositories' GitHub information.
+        Fetches a repository's GitHub information.
 
         The repository should look like `user/reponame` or `user reponame`.
+        If it's not a stored repo or PyDis repo, it will fetch the most-starred repo
+        matching the search query from GitHub.
         """
-        repo = "/".join(repo)
-        if repo.count("/") != 1:
-            embed = discord.Embed(
-                title=random.choice(NEGATIVE_REPLIES),
-                description="The repository should look like `user/reponame` or `user reponame`.",
-                colour=Colours.soft_red
-            )
-
-            await ctx.send(embed=embed)
-            return
+        is_pydis = False
+        fetch_most_starred = False
+        repo_query = "/".join(repo)
+
+        # Determine type of repo
+        if repo_query.count("/") != 1:
+            if repo_query in self.stored_repos:
+                repo_query = self.stored_repos[repo_query]
+            else:
+                for each in self.repos:
+                    if repo_query == each["name"]:
+                        repo_query = each["full_name"]
```

Comment:
```suggestion
                        repo_query = each
```
With this this change, you should be able to avoid the `for-loop` in `Case 1`, and maybe add comment to clarify:
```py
if is py_dis:
   repo_data = repo_query # Earlier, we already matched the repo with 'repo_query = each'
```
Techinically you could also do `repo_data = each`, and then then it would be `if fetch_most_starred`... `elif not is_pydis` and therefore removing the extra `if is_pydis` check but that might hurt readability?